messaging/src/main/java/org/axonframework/commandhandling/SimpleCommandHandlingComponent.java

```diff
@@ -82,39 +81,33 @@ public SimpleCommandHandlingComponent subscribe(@Nonnull QualifiedName name,
 
     @Override
     public SimpleCommandHandlingComponent subscribe(@Nonnull CommandHandlingComponent commandHandlingComponent) {
-        requireNonNull(commandHandlingComponent, "The command handling component may not be null");
-        subComponents.add(commandHandlingComponent);
+        subComponents.add(requireNonNull(commandHandlingComponent, "The command handling component may not be null"));
         return this;
     }
 
     @Nonnull
     @Override
     public MessageStream.Single<? extends CommandResultMessage<?>> handle(@Nonnull CommandMessage<?> command,
                                                                           @Nonnull ProcessingContext context) {
-        QualifiedName qualifiedName = command.type().qualifiedName();
-        // TODO #3103 - add interceptor knowledge
-        Optional<CommandHandlingComponent> optionalSubHandler = subComponents
-                .stream()
-                .filter(subComponent ->
-                                subComponent.supportedCommands().contains(qualifiedName)
-                )
-                .findFirst();
+        QualifiedName qualifiedName = requireNonNull(command, "The command message cannot be null.")
```

Comment:
Like this is fine though for me. It chains, but is not used in another method call's arguments

Reply:
Gotcha, makes sense. I'll keep this one around.